### ISSUE TITLE
users can't delete own links

### ISSUE THREAD

**Issue by gitthismoney** [cydrobolt/polr#160]
Hello,

I noticed that regular users can't delete their own links... is there any way to enable or implement this? 

Thanks!

**Comment by cydrobolt** [comment]
This is by design because allowing regular users to delete their own links can allow them to abuse your app. For instance, they could practically edit their link destination. They can create a link to a good destination, then delete it and recreate it with a malicious destination. I don't think it is a wise idea to provide this option because of how easily it can be abused. 

If you want, you can always edit the code and implement it yourself if you want.

**Comment by brainchild0** [comment]
I started using Polr a few days ago.  I am glad that someone has implemented an open URL shortener application that has a clean interface and accessible feature set.  Alternatives tend to have a greater number of features.  Some of these features are useful, but many others contribute to a bloated feel and distracting interface.   (Of course, some new features could be considered for this project that would not add bloat, if any potential maintainers are interested in carrying them forward.)

Soon after beginning use, however, I too noticed the limitation addressed in this issue.  I too feel that it would be very useful and appropriate to enable users to delete their own links.

As administrators are currently allowed to delete links, whereas other users are not, perhaps one solution would be to add finer granularity to the set of user roles, introducing the possibility that some users are delegated permission to delete their own links, but not to add or to remove other users.

The more fundamental question, though, is the reason for  the design choice to disallow this case even for some users.  I am puzzled by the explanation given.  Some links are malicious, surely, and some people are unscrupulous, surely.  Any address may in principle be malicious, whether it was the original destination of some link, or one assigned after the link was created with a different original destination.  It is the job of a user who might visit a destination, and of client software, to make this discrimination.   I have failed to understand how allowing this use case would introduce additional danger or abuse potential to the administrator of an instance of the application.  The suggestion that it should be disallowed to delete a link and then to recreate one of the same name appears to be in direct conflict with the plain observation that *simply editing a link destination currently is allowed, for all users.*

I am happy to consider that I may have missed some critical problem, but after reviewing the discussion, I  suggest  reopening this request.
